### PR TITLE
Fixed #15799 -- Doc'd that Storage._open() should raise FileNotFoundError when file doesn't exist.

### PR DESCRIPTION
**Pull request by aokugel**
Section in documentation:

https://docs.djangoproject.com/en/4.2/howto/custom-file-storage/#django.core.files.storage._open

Thanks!

### CODE REVIEW

File: docs/howto/custom-file-storage.txt [hunk 1]
```diff
@@ -138,4 +144,4 @@ Storages are then accessed by alias from from the
 
     from django.core.files.storage import storages
 
-    example_storage = storages["example"]
+    example_storage = storages["example"]
```
Comment (felixxm) [inline docs/howto/custom-file-storage.txt:147]: Revert unrelated change.

File: docs/howto/custom-file-storage.txt [hunk 2]
```diff
@@ -76,7 +76,13 @@ objects. These are:
...
 uses to open the file. This must return a ``File`` object, though in most cases,
 you'll want to return some subclass here that implements logic specific to the
-backend storage system.
+backend storage system. This will raise a ``FileNotFoundError`` exception if the
+file does not exist. To catch: ::
+
+    try:
+        f = myStorage.open()
+    except FileNotFoundError as e:
+        print(f"File {path} not found!", file=sys.stderr)
```
Comment (felixxm) [inline docs/howto/custom-file-storage.txt:85]: This ticket is about an exception that should be raised by `_open()`, not how to catch exceptions from `open()`. Maybe:
```suggestion
backend storage system. The :exc:`FileNotFoundError` exception should be raised
when a file doesn't exist.
```

**Review by felixxm — COMMENTED** [review]
@aokugel Thanks :+1: I left comments.

File: docs/howto/custom-file-storage.txt (hunk 1 above)
Comment (aokugel) [reply docs/howto/custom-file-storage.txt:147]: I originally added to the end of the file, deleted what I had, but left a trailing empty line by accident.  I then removed the extra line in a later commit.  The code has not been modified in any way.

File: docs/howto/custom-file-storage.txt (hunk 2 above)
Comment (aokugel) [reply docs/howto/custom-file-storage.txt]: Thank you for your feedback.  I misunderstood the original ask.  I have reflected your feedback in the most recent commit.

**Comment by felixxm** [comment]
@aokugel Thanks for updates :+1:
